src/mc_rbdyn/Robot.cpp

```diff
@@ -527,18 +527,23 @@ Eigen::Vector3d Robot::comAcceleration() const
 
 sva::ForceVecd Robot::bodyWrench(const std::string & bodyName) const
 {
-  const auto & fs = bodyForceSensor(bodyName);
+  const auto & fs = findBodyForceSensor(bodyName);
   sva::ForceVecd w_fsactual = fs.wrenchWithoutGravity(*this);
-  sva::PTransformd X_fsactual_surf = fs.X_fsactual_parent();
+  const auto & X_0_body = bodyPosW(bodyName);
+  const auto & X_0_parent = bodyPosW(fs.parentBody());
+  const auto X_parent_body = X_0_body * X_0_parent.inv();
+  sva::PTransformd X_fsactual_surf = X_parent_body * fs.X_fsactual_parent();
```

Comment:
The extra lookup/computation are completely useless if the sensor is directly attached to the body (common case). I think it's worth checking `hasForceSensor` to bypass that in the common case

Reply:
Indeed that's better.